qiskit/circuit/random/utils.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility functions for generating random circuits."""
+
+from inspect import signature
+import numpy as np
+
+from qiskit.circuit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.circuit import Reset
+from qiskit.extensions import (IdGate, U1Gate, U2Gate, U3Gate, XGate,
+                               YGate, ZGate, HGate, SGate, SdgGate, TGate,
+                               TdgGate, RXGate, RYGate, RZGate, CnotGate,
+                               CyGate, CzGate, CHGate, CrzGate, Cu1Gate,
+                               Cu3Gate, SwapGate, RZZGate,
+                               ToffoliGate, FredkinGate)
+from qiskit.exceptions import QiskitError
+
+
+def random_circuit(n_qubits, depth, max_operands=3, measure=False,
+                   conditional=False, reset=False, seed=None):
+    """Generate random circuit of arbitrary size and form.
+
+    Args:
+        n_qubits (int): number of quantum wires
+        depth (int): layers of operations (i.e. critical path length)
+        max_operands (int): maximum operands of each gate (between 1 and 3)
+        measure (bool): if True, measure all qubits at the end
+        conditional (bool): if True, insert middle measurements and conditionals
```

Comment:
I don't think we end up generating middle measurements, but that's not blocking.